modules/registrars/openprovider/Controllers/Hooks/DomainLockingEnabledController.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace OpenProvider\WhmcsRegistrar\Controllers\Hooks;
+
+use OpenProvider\API\ApiHelper;
+use OpenProvider\WhmcsRegistrar\helpers\DomainFullNameToDomainObject;
+use WHMCS\Database\Capsule;
+use WeDevelopCoffee\wPower\Models\Domain;
+use Openprovider\Api\Rest\Client\Domain\Api\DomainServiceApi;
+use OpenProvider\WhmcsRegistrar\helpers\Cache;
+
+/**
+ * Class DomainController
+ * OpenProvider Registrar module
+ *
+ * @copyright Copyright (c) Openprovider 2018
+ */
+class DomainLockingEnabledController
+{
+
+    /**
+     * @var ApiHelper
+     */
+    private $apiHelper;
+    /**
+     * @var Domain
+     */
+    private $domain;
+
+    /**
+     * DomainController constructor.
+     * @param ApiHelper $apiHelper
+     */
+    public function __construct(ApiHelper $apiHelper, Domain $domain)
+    {
+        $this->apiHelper = $apiHelper;
+        $this->domain = $domain;
+    }
+
+    public function handleDomainLockingClientArea($vars)
+    {
+        $id = $vars['domain']->id ?? null;
```

Comment:
don;t we need check domain exists in the $vars

Reply:
Added a check to check domain existance in the $vars